tests/hamt.rs

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_hamt::*;
+use microkelvin::{Annotation, Cardinality, Child, Compound, Nth};
+
+fn correct_empty_state<C, A>(c: C) -> bool
+where
+    C: Compound<A>,
+    A: Annotation<C::Leaf>,
+{
+    for i in 0.. {
+        match c.child(i) {
+            Child::EndOfNode => return true,
+            Child::Empty => (),
+            _ => return false,
+        }
+    }
+    unreachable!()
+}
+
+#[test]
+fn trivial() {
+    let mut hamt = Hamt::<u32, u32, ()>::new();
+    assert_eq!(hamt.remove(&0).unwrap(), None);
```

Comment:
Same as the other PR: make the tests returns a `Result` and use `?`, instead of using `unwrap`s, and/or use `expect`s whenever this is not enough.

This is applying to the whole tests.